src/libraries/System.Net.Http/src/System/Net/Http/SocketsHttpHandler/HttpConnectionPoolManager.cs

```diff
@@ -517,6 +535,14 @@ private void RemoveStalePools()
             // be returned to pools they weren't associated with.
         }
 
+        private void HeartBeat()
+        {
+            foreach (HttpConnectionPool pool in _pools.Values)
+            {
+                pool.HeartBeat();
+            }
```

Comment:
`ConcurrentDictionary.Values` locks the collection and copies all the values to a list -- not good to use here.

Enumerating the full collection acquires no locks.

```suggestion
            foreach (KeyValuePair<HttpConnectionKey, HttpConnectionPool> pool in _pools)
            {
                pool.Value.HeartBeat();
            }
```